ts/screens/messages/PaginatedMessagesHomeScreen.tsx

```diff
@@ -0,0 +1,290 @@
+ */
+class MessagesHomeScreen extends React.PureComponent<Props, State> {
+  navListener?: NavigationEventSubscription;
+
+  animatedTabScrollPositions: ReadonlyArray<Animated.Value> = [
+    new Animated.Value(0),
+    new Animated.Value(0),
+    new Animated.Value(0)
+  ];
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      currentTab: 0
+    };
+  }
+
+  componentDidMount() {
+    this.props.loadNextPage();
+    // eslint-disable-next-line functional/immutable-data
+    this.navListener = this.props.navigation.addListener("didFocus", () => {
+      setStatusBarColorAndBackground(
+        "dark-content",
+        customVariables.colorWhite
+      );
+    });
+  }
+
+  componentWillUnmount() {
+    if (this.navListener) {
+      this.navListener.remove();
+    }
+  }
+
+  // It create a mostly 2 states output: it value is mostly 0 or HEADER_HEIGHT
+  getHeaderHeight = (): Animated.AnimatedInterpolation =>
+    this.animatedTabScrollPositions[this.state.currentTab].interpolate({
+      inputRange: [0, HEADER_HEIGHT],
+      outputRange: [0, 1],
+      extrapolate: "clamp"
+    });
+
+  render() {
+    const {
+      isSearchEnabled,
+      messagesState,
+      navigateToMessageDetail,
+      loadNextPage,
+      loadPreviousPage,
+      servicesById,
+      paymentsByRptId,
+      updateMessagesArchivedState
+    } = this.props;
+
+    const { error, isLoading, allMessages, nextCursor, previousCursor } =
+      unwrapMessageState(messagesState);
+
+    return (
+      <TopScreenComponent
+        accessibilityEvents={{
+          disableAccessibilityFocus:
+            this.props.messageSectionStatusActive !== undefined
+        }}
+        accessibilityLabel={I18n.t("messages.contentTitle")}
+        contextualHelpMarkdown={contextualHelpMarkdown}
+        faqCategories={["messages"]}
+        headerTitle={I18n.t("messages.contentTitle")}
+        isSearchAvailable={{ enabled: true, searchType: "Messages" }}
+        appLogo={true}
+      >
+        <SectionStatusComponent
+          sectionKey={"messages"}
+          onSectionRef={v => {
+            setAccessibilityFocus(v, 100 as Millisecond);
+          }}
+        />
+        {!isSearchEnabled && (
+          <React.Fragment>
+            <AnimatedScreenContentHeader
+              title={I18n.t("messages.contentTitle")}
+              iconFont={{ name: "io-home-messaggi", size: MESSAGE_ICON_HEIGHT }}
+              dynamicHeight={this.getHeaderHeight()}
+            />
+            <HomeTabs
+              error={error}
+              isLoading={isLoading}
+              allMessages={allMessages}
+              hasNextPage={!!nextCursor}
+              loadNextPage={() => {
+                if (nextCursor) {
+                  loadNextPage(nextCursor);
+                }
+              }}
+              loadPreviousPage={() => {
+                if (previousCursor) {
+                  loadPreviousPage(previousCursor);
+                }
+              }}
```

Comment:
You definitely got me here!
Now, as per our chat:

1. there is a new state tracking [the last request](https://github.com/pagopa/io-app/pull/3441/files#diff-482523dcf54c84f4e40e09c47ee2eeca62c92423ef82ff98aae91839199d8c16R31)
2. because of (1) the app can now determine whether we are loading previous or next messages
3. the responsibiltiy for (2) is encapsulated in [MessageList](https://github.com/pagopa/io-app/pull/3441/files#diff-a0ab177007bd8f5d749d9e7494df01aaddde642f3d379c200e99e9c6048b9b17R147) and its ancestors are completely oblivious of it

In the screenshot below, you can see that only one request is made even if the `onEndReached` event was fired.

<img width="1937" alt="Screenshot 2021-11-03 at 08 56 48" src="https://user-images.githubusercontent.com/2094604/140031086-5006a65e-f062-4a57-952e-4eab2dd54578.png">